nodes/src/nodes/tool_exa_search/exa_driver.py

```
@@ -0,0 +1,338 @@
+                    'Returns structured results with title, URL, text content, relevance score, and published date. '
+                    'Optional: "num_results", "type" (auto/neural/keyword), "use_autoprompt", '
+                    '"include_domains", "exclude_domains", "start_published_date", "end_published_date", "include_text".'
+                ),
+                'inputSchema': INPUT_SCHEMA,
+            }
+        ]
+
+    def _tool_validate(self, *, tool_name: str, input_obj: Any) -> None:  # noqa: ANN401
+        bare = self._bare_name(tool_name)
+        if bare != 'exa_search':
+            raise ValueError(f'Unknown tool {tool_name!r} (expected exa_search)')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)